### PR TITLE
Finalise contract

### PR DESCRIPTION
**Pull request by willjgriff**
Added max size for Redeemable Tokens array, currently set to 30. 

Renamed some stuff for clarity including a couple of function names which I've updated everywhere I think they need to be.

Also should the app be initialised with some redeemable tokens? It currently isn't, don't know if it would be useful or not.

### CODE REVIEW

File: contracts/Redemptions.sol [hunk 1]
```diff
@@ -51,71 +52,73 @@ contract Redemptions is AragonApp {
...
-        emit RemoveToken(_token);
+        emit RemoveRedeemableToken(_token);
     }
 
     /**
-    * @dev Redeem function is intended to only be used directly, using a forwarder will not work see: https://github.com/1Hive/redemptions-app/issues/78
-    * @notice Redeem `@tokenAmount(self.token(): address, _amount, true)` tokens
-    * @param _amount amount of tokens
+    * @dev The redeem function is intended to be used directly, using a forwarder will not work see: https://github.com/1Hive/redemptions-app/issues/78
+    * @notice Burn `@tokenAmount(self.burnableToken(): address, _amount, true)` in exchange for redeemable tokens.
```
Comment (fabriziovigevani) [inline contracts/Redemptions.sol]:
```suggestion
    * @notice Burn `@tokenAmount(self.burnableToken(): address, _burnableAmount, true)` in exchange for redeemable tokens.
```

File: docs/technical-documentation.md [hunk 2]
```diff
@@ -12,45 +12,45 @@ The Redemptions app must have the `TRANSFER_ROLE` permission on `_vault` and the
...
 
 ## Removing Tokens
 
-Removing tokens from the Redemptions app is done by passing an address `_token` to the removeToken() function. This must be an address that is already added to the Redemptions `tokenAdded[]` array.
+Removing tokens from the Redemptions app is done by passing an address `_token` to the removeRedeemableToken() function. This must be an address that is already added to the Redemptions `redeemableTokenAdded` mapping.
 ```
-function removeToken(address _token) external auth(REMOVE_TOKEN_ROLE) {
-	require(tokenAdded[_token], ERROR_NOT_VAULT_TOKEN);
+function removeRedeemableToken(address _token) external auth(REMOVE_TOKEN_ROLE) {
+	require(redeemableTokenAdded[_token], ERROR_NOT_VAULT_TOKEN);
```
Comment (fabriziovigevani) [inline docs/technical-documentation.md]:
```suggestion
	require(redeemableTokenAdded[_token], ERROR_TOKEN_NOT_ADDED);
```

File: contracts/examples/Template.sol [hunk 3]
```diff
@@ -85,7 +85,7 @@ contract Template is TemplateBase {
         // Initialize apps
         vault.initialize();
         tokenManager.initialize(token, true, 0);
-        redemptions.initialize(vault, tokenManager);
+        redemptions.initialize(vault, tokenManager, new address[](0));
```
Comment (rperez89) [inline contracts/examples/Template.sol]: i think that we need to add the redeemable token RDT to that list right @fabriziovigevani ?

Comment (fabriziovigevani) [reply contracts/examples/Template.sol]: No, the RDT token would be the `tokenManager`. 
The list is for redeemable tokens (eth, dai, etc)

**Pull request merged**